src/main/scala/org/constellation/util/Rewards.scala

```diff
@@ -0,0 +1,72 @@
+package org.constellation.util
+
+object Rewards {
+  val roundingError = 0.000000000001
+
+  val epochOne = 438000
+  val epochTwo = 876000
+  val epochThree = 1314000
+  val epochFour = 1752000
+
+  val epochOneRewards = 0.68493150684//10k/mo, 328.767123287/day
+  val epochTwoRewards = 0.34246575342
+  val epochThreeRewards = 0.17123287671
+  val epochFourRewards = 0.08561643835
+
+  /*
+  Partitioning of address space, light nodes have smaller basis that full. Normalizes rewards based on node size
+   */
+  val partitonChart = Map[String, Set[String]]()
+
+  /*
+  Should come from reputation service
+   */
+  val transitiveReputationMatrix = Map[String, Map[String, Double]]()
+  val neighborhoodReputationMatrix = Map[String, Double]()
+
+  /*
+  snapshots - 10yrs @ ~20/hr
+   */
+  val rewardsPool = 1752000
+
+  def validatorRewards(
+                        curShapshot: Int,
+                       transitiveReputationMatrix: Map[String, Map[String, Double]],
+                       neighborhoodReputationMatrix: Map[String, Double],
+                       partitonChart: Map[String, Set[String]]
+                      )= {
+    val trustEntropyMap = shannonEntropy(transitiveReputationMatrix, neighborhoodReputationMatrix)
+    val distro = rewardDistribution(partitonChart, trustEntropyMap)
+    distro.mapValues(_ * rewardForEpoch(curShapshot))
+  }
+
+  def rewardForEpoch(curShapshot: Int) = curShapshot match {
+    case num if num >= 0 && num < epochOne  => epochOneRewards
+    case num if num >= 438000 && num < epochTwo => epochTwoRewards
+    case num if num >= 876000 && num < epochThree => epochThreeRewards
+    case num if num >= 1314000 && num < epochFour => epochFourRewards
+    case _ => 0d
+  }
+
+  def shannonEntropy(
+                      transitiveReputationMatrix: Map[String, Map[String, Double]],
+                      neighborhoodReputationMatrix: Map[String, Double]
+                    ) = {
+    val weightedTransitiveReputation = transitiveReputationMatrix.map { case (key, view) =>
+      val neighborView = view.map{ case (neighbor, score) => neighborhoodReputationMatrix(key) * score }.sum
+        (key, neighborView)
+    }
+    weightedTransitiveReputation.mapValues{ trust => - trust * math.log(trust)/math.log(2) }
```

Comment:
assert trust > 0
?

Reply:
Hey Nikolaij, reputation and partitioning are mocked for now and will be tested independently. Also a trust score of 0 is possible.

Reply:
@buckysballs Mhm, maybe the comment should have been more explicit: If the `math.log` doesn't do it on it's own, there should definately be a debug check or assert (independent of whether it's intended for the double to be negative or not)

Reply:
You're right we need to map 0 log 0 -> 0 thanks!

Reply:
Also noodled on this. We shouldn't allow 0.0 as a rep score since that will mess with the normalization. Will add an offset.

Reply:
@buckysballs Which normalization in particular and how will it mess with it? If it's a plain additive normalization, a value of 0 should work, no? 
If You start out with non-zero trust values, I imagine there are a lot of ways to change the trust values that make it so they never actually go to zero. I'd have to see where an offset-solution comes in, in a way that it doesn't make the calculuations more complitacted (magic numbers determining edge behaviour). Another magic number way of doing things would be through epsilons instead of boundaries.
You may also just abstract `- trust * math.log(trust)/math.log(2)` into a function and observe how it behaves in a bunch of text. (I don't know the context in which the entropy function does its magic here, but possibly it's nice features are anyway for nil if there's too much correlations introduced in various corners of the system and this pretty log would do the same job as many other functions.)